src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ObjectNodeSection.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    [Flags]
+    public enum CustomSectionAttributes
```

Comment:
Nit: This enum is a piece of ABI with ObjectWriter. We have been generally avoiding using ABI enums and structs within the rest of the compiler (e.g. we have a bunch of translations from compiler enums to JIT enums within CorInfoImpl.cs). Many of the enums are blittable and it looks silly, but not having the backends hardcoded lets us have different codegens and different object writers.